### ISSUE TITLE
update_pars incompatible with subclasses

### ISSUE THREAD

**Issue by daniel-klein** [starsimhub/starsim#545]
Within modules.py, the `update_pars` class assumes that all user pars should match a default. Any additional pars raise a ValueError

```python
# Should be no remaining pars
        if len(pars):
            errormsg = f'{len(pars)} unrecognized arguments for {self.name}: {sc.strjoin(pars.keys())}'
            raise ValueError(errormsg)
```

But what if we are creating a derived class that has additional parameters? Consider:
```python
class Zombie(ss.SIR):
    def __init__(self, pars=None, **kwargs):
        super().__init__(**kwargs)

        self.default_pars(
            inherit = True, # Inherit from SIR defaults
            p_fast = ss.bernoulli(p=0.10), # Fast zombies attack more people!
        )
        self.update_pars(pars, **kwargs)
```
If the user tries to override the default for p_fast, `super().__init__(**kwargs)` raises the value error .

A workaround is to not pass **kwargs into the initialization of the parent class, `super().__init__()`, but this could have unintended consequences later.

Might be better to raise a warning instead of a ValueError or otherwise handle it like the `inherit` flag?

**Comment by cliffckerr** [comment]
`super().__init__()` should indeed not take any input arguments. Arguments should only be passed to `update_pars()`.

Previously, we passed arguments to _both_ `__init__()` and `update_pars()`, and misspelled arguments silently failed! This issue came up in the APHRC ABM training, which is why we changed. You definitely need to pass arguments to `update_pars`. But you _shouldn't_ need to pass any arguments to `__init__()` -- the only things that should need to get passed to the parent should be handled by the pars.

It's true that there are some limitations of this approach, but it seems to follow the "principle of least surprise". I also personally like the simplicity of `__init__()` not taking any arguments, because there have been some bugs in previous Starsim models around handling this.
